File: android/src/main/java/com/reactnativechimesdk/itf/SimpleAudioVideoObserver.java
package com.reactnativechimesdk.itf;

import android.util.Log;

import androidx.annotation.NonNull;

import com.amazonaws.services.chime.sdk.meetings.audiovideo.AudioVideoObserver;
import com.amazonaws.services.chime.sdk.meetings.session.MeetingSessionStatus;

import java.util.Objects;

public interface SimpleAudioVideoObserver extends AudioVideoObserver {

  String TAG = "SimpleAudioVideoObserve";

  @Override
  default void onAudioSessionCancelledReconnect() {

  }

  @Override
  default void onAudioSessionDropped() {

  }

  @Override
  default void onAudioSessionStarted(boolean b) {
  }

  @Override
  default void onAudioSessionStartedConnecting(boolean b) {

  }

  @Override
  default void onAudioSessionStopped(@NonNull MeetingSessionStatus meetingSessionStatus) {
    Log.d(TAG, "onAudioSessionStopped: " + Objects.requireNonNull(meetingSessionStatus.getStatusCode()).name());
  }

  @Override
  default void onConnectionBecamePoor() {

  }

  @Override
  default void onConnectionRecovered() {

  }

  @Override
  default void onVideoSessionStarted(@NonNull MeetingSessionStatus meetingSessionStatus) {
    Log.d(TAG, "onVideoSessionStarted: " + Objects.requireNonNull(meetingSessionStatus.getStatusCode()).name());
  }

  @Override
  default void onVideoSessionStartedConnecting() {

  }

  @Override
  default void onVideoSessionStopped(@NonNull MeetingSessionStatus meetingSessionStatus) {
    Log.d(TAG, "onVideoSessionStopped: " + Objects.requireNonNull(meetingSessionStatus.getStatusCode()).name());
  }
}

File: android/src/main/java/com/reactnativechimesdk/itf/SimpleAudioVideoObserver.kt
package com.reactnativechimesdk.itf

import android.util.Log
import com.amazonaws.services.chime.sdk.meetings.audiovideo.AudioVideoObserver
import com.amazonaws.services.chime.sdk.meetings.session.MeetingSessionStatus

interface SimpleAudioVideoObserver : AudioVideoObserver {

  companion object {
    const val TAG = "SimpleAudioListener"
  }

  override fun onAudioSessionCancelledReconnect() {
    Log.d(TAG, "onAudioSessionCancelledReconnect: ")
  }

  override fun onAudioSessionDropped() {
    Log.d(TAG, "onAudioSessionDropped: ")
  }

  override fun onAudioSessionStarted(reconnecting: Boolean) {
    Log.d(TAG, "onAudioSessionStarted: ")
  }

  override fun onAudioSessionStartedConnecting(reconnecting: Boolean) {
    Log.d(TAG, "onAudioSessionStartedConnecting: ")
  }

  override fun onAudioSessionStopped(sessionStatus: MeetingSessionStatus) {
    Log.d(TAG, "onAudioSessionStopped: ")
  }

  override fun onConnectionBecamePoor() {
    Log.d(TAG, "onConnectionBecamePoor: ")
  }

  override fun onConnectionRecovered() {
    Log.d(TAG, "onConnectionRecovered: ")
  }

  override fun onVideoSessionStarted(sessionStatus: MeetingSessionStatus) {
    Log.d(TAG, "onVideoSessionStarted: ")
  }

  override fun onVideoSessionStartedConnecting() {
    Log.d(TAG, "onVideoSessionStartedConnecting: ")
  }

  override fun onVideoSessionStopped(sessionStatus: MeetingSessionStatus) {
    Log.d(TAG, "onVideoSessionStopped: ")
  }
}
